WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/DashboardDomainRegistrationCardCell.swift

```diff
@@ -0,0 +1,110 @@
+import UIKit
+import WordPressFlux
+
+final class DashboardDomainRegistrationCardCell: BaseDashboardDomainsCardCell {
+
+    // MARK: - View Model
+
+    override var viewModel: DashboardDomainsCardViewModel {
+        return cardViewModel
+    }
+
+    private lazy var cardViewModel: DashboardDomainsCardViewModel = {
+        let onViewTap: () -> Void = { [weak self] in
+            self?.cardTapped()
+        }
+        let onHideThisTap: UIActionHandler = { [weak self] _ in
+            self?.hideCardTapped()
+        }
+        return DashboardDomainsCardViewModel(
+            strings: .init(
+                title: Strings.title,
+                description: Strings.content,
+                hideThis: Strings.hideThis,
+                source: Strings.source
+            ),
+            onViewTap: onViewTap,
+            onHideThisTap: onHideThisTap
+        )
+    }()
+
+    // MARK: - User Interaction
+
+    private func cardTapped() {
+        guard let props = makeUnwrappedProperties() else {
+            return
+        }
+        WPAnalytics.track(.domainCreditRedemptionTapped)
+        DomainsDashboardCoordinator.presentDomainsSuggestions(
+            in: props.presentingViewController,
+            source: Strings.source,
+            blog: props.blog
+        )
+    }
+
+    private func hideCardTapped() {
+        guard let props = makeUnwrappedProperties() else {
+            return
+        }
+        let service = BlogDashboardPersonalizationService(siteID: props.siteID.intValue)
+        service.setEnabled(false, for: .domainRegistration)
+    }
+
+    // MARK: - Constants
+
+    private static var hasLoggedDomainCreditPromptShownEvent: Bool = false
+
+    private enum Strings {
+        static let title = NSLocalizedString(
+            "Register Domain",
+            comment: "Action to redeem domain credit."
+        )
+        static let content = NSLocalizedString(
+            "All WordPress.com plans include a custom domain name. Register your free premium domain now.",
+            comment: "Information about redeeming domain credit on site dashboard."
+        )
+        static let hideThis = NSLocalizedString(
+            "domain.dashboard.card.menu.hide",
+            value: "Hide this",
+            comment: "Title for a menu action in the context menu on the Jetpack install card."
+        )
+        static let source = "domain_registration_dashboard_card"
+    }
```

Comment:
Nitpick: Moving these Strings out of the way (to maybe end of the file as an extension) could benefit readability. They can bloat become something you scroll through.

Reply:
@alpavanoglu I Just realized I'd added these strings to the middle of the file.  I usually add to the bottom of the file. Thanks for pointing this out.